src/category_theory/monoidal/category.lean

```diff
@@ -600,6 +600,40 @@ end
 
 end
 
+section
+
+universes v₁ v₂ u₁ u₂
+
+variables (C₁ : Type u₁) [category.{v₁} C₁] [monoidal_category.{v₁} C₁]
+variables (C₂ : Type u₂) [category.{v₂} C₂] [monoidal_category.{v₂} C₂]
+
+local attribute [simp]
+associator_naturality left_unitor_naturality right_unitor_naturality pentagon
+
+@[simps]
+instance prod_monoidal : monoidal_category (C₁ × C₂) :=
+{ tensor_obj := λ X Y, (X.1 ⊗ Y.1, X.2 ⊗ Y.2),
+  tensor_hom := λ _ _ _ _ f g, (f.1 ⊗ g.1, f.2 ⊗ g.2),
+  tensor_unit := (𝟙_ C₁, 𝟙_ C₂),
+  associator := λ X Y Z, (α_ X.1 Y.1 Z.1).prod (α_ X.2 Y.2 Z.2),
+  left_unitor := λ X,
+  begin
+    dsimp,
+    have : X = (X.1, X.2) := prod.ext rfl rfl,
+    rw this,
+    dsimp,
+    exact (λ_ X.1).prod (λ_ X.2),
+  end,
+  right_unitor := λ X,
+  begin
+    dsimp,
+    have : X = (X.1, X.2) := prod.ext rfl rfl,
+    rw this,
+    exact (ρ_ X.1).prod (ρ_ X.2),
+  end }
```

Comment:
There is something I don't quite understand here.

I tried to write `left_unitor` and `right_unitor` similarly to `associator`, e.g.,
```
left_unitor := λ X, (λ_ X.1).prod (λ_ X.2)
```
but I'm getting a type mismatch and I need to do extra work to convince the type checker.  If I write
```
left_unitor := λ ⟨X, Y⟩, (λ_ X).prod (λ_ Y)
```
then this seems to produce worse simp lemmas and in the definition of `prod` for lax monoidal functors the automation cannot fill in the necessary proofs.  I tried to keep the shorter definition of the unitors but add explicit simplification lemmas like
```
@[simp] lemma prod_monoidal_left_unitor_hom (X : C₁ × C₂) :
  (λ_ X).hom = ((λ_ X.1).hom, (λ_ X.2).hom) := by { cases X, refl }
```
but for some reason that didn't help.

@semorrison I would appreciate your help debugging this.

Reply:
I've had some luck with removing the `@[simps]` attribute and writing all simp projection lemmas manually.  It seems that when the automation tries to fill in the proofs in the definition of `prod` of lax monoidal fucntors, the automatically generated lemmas for the unitors are applied first and get the simplifier stuck.  I don't think I can tell with confidence I understand how exactly the simplifier works and how simp lemmas are chosen or in which order they are tried... but removing `@[simps]` and writing the simp lemmas I want seems to allow me to keep the definitions of the unitors nice and simple and have the proofs filled in automatically in other places.  Is there a way to selectively override projections?  Or perhaps remove the `@[simp]` attribute from a generated lemma to remove it from the simplifier database?  (I feel I've seen something like this in code but can't find.)

Reply:
You can write `@[simps associator]` if you just want that one simp lemma.

Reply:
Thank you, that's helpful!  I've generated some of the projections and provided the rest manually.

Reply:
I think this is fine, but I'll just ping @fpvandoorn (who provided us with `simps`) in case he's interested to see this corner case.